modules/standard/Buffers.chpl

```diff
@@ -76,50 +77,65 @@ module BufferInternals {
 
 }
 
+/* Support for buffers - regions of memory without a particular interpretation.
+   
+   This module provides :record:`bytes` and :record:`buffer` types which
+   can be used to manage memory regions.
+
+   A :record:`bytes` is a contiguous memory region - really a C
+   pointer, length, along with information necessary to free
+   the memory region when it is no longer used.
+
+   A :record:`buffer` consists of a sequence views into :record:`bytes`
+   objects. A :record:`bytes` object might be shared by several
+   :record:`buffer` objects. 
+
+   These types should be safe to use multi-locale. These types should free
+   their memory after the last user of that memory goes out of scope. They are
+   currently reference counted but that may not always be the case.
+ */
 module Buffers {
   use BufferInternals; // TODO -- non-exporting use
 
   // Now define the Chapel types using the originals..
-  record bytes {
-    var home: locale;
-    var _bytes_internal:qbytes_ptr_t;
-  }
 
-  record buffer_iterator {
+  /* This type represents a contiguous sequence of bytes.
+     This sequence of bytes is represented with a C
+     pointer and length and is currently reference counted. */
```

Comment:
Maybe you should add "it also contains some private fields" here, so that people unfamiliar with the concept of a locale don't get strange ideas about what it does